llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -12074,6 +12078,12 @@ SDValue DAGCombiner::visitSELECT(SDNode *N) {
 
     if (SDValue NewSel = SimplifySelect(DL, N0, N1, N2))
       return NewSel;
+
+    // (select (ugt x, C), (add x, ~C), x) -> (umin (add x, ~C), x)
```

Comment:
Is there a version of this pattern with `ult` and the `select` operands swapped?

Reply:
I'll add it.

Reply:
Added. It stems from corresponding C code.